scenes/game/ai_player.gd

```diff
@@ -1,26 +1,82 @@
-extends Node2D
+## AI player controller object.
+
+## This file describes a bunch of hooks and basic functions used to provide an
+## AI with options, then relay the AI's choice from those options back to the
+## game controller. The actual AI decision-making is done within its
+## [i]policy[/i], which is a Node that implements the laundry list of pick_*
+## functions. See ai_random_policy.gd for a very basic (but complete!) example.
+
+## The player is initialized with a reference to the relevant game and player
+## objects. After that, the main entry point into the code is through [method take_turn],
+## which is called by game.gd's `ai_take_turn`. `take_turn` consults the stored
+## game state the game state and returns an instance of one of the various
+## *Action classes defined in this file. `ai_take_turn` then uses that return
+## value to manifest the appropriate changes in the game proper.
+
+## `take_turn` enumerates all legal actions for the AI to take, then passes them
+## to policy (and then forwards the return value back to game.gd as described
+## above). Note that each possible variant of a (game) action counts as a
+## distinct (AI) action; for example, the choice of which cards to discard for
+## Force generation during a Walk.
+
+## A broad overview of the main chunks of this file:
+
+##   - class *State: Data wrappers for parts of game state.
+##   - class *Action: Represents an action the AI may or chooses to take,
+##         including parameters like which cards to discard in payment.
+##   - func get_*_actions: List all possible Action objects representing
+##         the named game action.
+##   - func pay_*, pick_*: Entry points for AI decisions that are not at
+##         the top of a turn; for example, a mid-strike decision on movement.
+
+class_name AIPlayer
+extends Resource
 
 const TEST_PrepareOnly = false
 
 const LocalGame = preload("res://scenes/game/local_game.gd")
-const AIPolicyRandom = preload("res://scenes/game/ai/ai_policy_random.gd")
-const AIPolicyRules = preload("res://scenes/game/ai/ai_policy_rules.gd")
 const Enums = preload("res://scenes/game/enums.gd")
 const CardDatabase = preload("res://scenes/game/card_database.gd")
 
-var game_player : LocalGame.Player
-var game_state : AIGameState = AIGameState.new()
-
-var ai_policy = AIPolicyRules.new()
+var game_logic : LocalGame
+var game_state : AIGameState
+var ai_policy
+
+# TODO: Check if unused in production? Seems like this should just be
+# encapsulated in game_state.
+var game_player: LocalGame.Player
```

Comment:
It's used now because stuff isn't being passed in anymore. But yeah, once everything is wrapped up tidily in AIGameState, these should probably go away.